Add Weapon Offset Customization

Berthalamew opened the pull request.

Berthalamew committed:
Add extra graphics options

-Adds menu options for turning off the shader LODS and disabling the light suppressor.
-Disables 60fps cutscenes by default

Berthalamew committed:
whoops

remove unneeded things

Berthalamew committed:
Add Weapon Offset Customization

Press End to open menu.
Only includes magnum, smg, and sniper

Berthalamew committed:
Revert "whoops"

This reverts commit a3c508638c5c1f5a6eabfd1ebf4478bb801801ea.

Berthalamew committed:
Revert "Add extra graphics options"

This reverts commit b3e9861aceb900459cc1cea17b438e15f77d286d.

Berthalamew committed:
Fix compiling

Berthalamew committed:
Fixup menu

Berthalamew committed:
Create Separate Config for Weapon Offsets

-Saves WeaponOffsets.cfg in the same folder as the regular config

-Still need to setup the rest of the weapons

Berthalamew committed:
Label your stuff Kant

Berthalamew committed:
Add all weapons to offset menu

Berthalamew committed:
Fix bugs and setup offsets to apply on map load

Berthalamew committed:
Merge branch 'development-new' into development-new-weaponoffsets

Berthalamew committed:
Add stdafx header

Berthalamew committed:
Change header location

Berthalamew committed:
Add requested changes

-Change UI to use combobox instead of showing all offsets at once
-Remove END hotkey to open menu and add button to advanced hud settings menu to open it instead
-Remove duplicate code
-Add Spanish Translation

Berthalamew committed:
Merge branch 'development-new' into development-new-weaponoffsets

Berthalamew committed:
Fix conflicts

Berthalamew committed:
Don't store datum for weapon tags

nukeulater committed:
Merge branch 'development-new' into development-new-weaponoffsets

Berthalamew committed:
Cleanup Weapon Offset Code

- Cleanup code for offsets
- fix issue with deadzones not being applied

Berthalamew committed:
fixup

Berthalamew committed:
Move Weapon Offset data into a struct

Berthalamew committed:
Offset code cleanup

- fix bug with custom controller layout not being set in singleplayer

Berthalamew committed:
More Weapon Offset cleanup

Berthalamew committed:
Update WeaponOffsets.cpp

nukeulater commented on xlive/H2MOD/Modules/Shell/Config.cpp:
```diff
@@ -48,8 +48,8 @@ bool H2Config_mouse_uniform = false;
 H2Config_Deadzone_Type H2Config_Controller_Deadzone = H2Config_Deadzone_Type::Axial;
-float H2Config_Deadzone_A_X = 1.0f;
-float H2Config_Deadzone_A_Y = 1.0f;
+float H2Config_Deadzone_A_X = 26.0f;
```
any reason for modifying this?

Berthalamew replied:
The default deadzone ingame is 26, for some reason it's set to 1 in the config so I changed it to match that

nukeulater commented on xlive/H2MOD/Modules/WeaponOffsets/WeaponOffsetConfig/WeaponOffsetConfig.cpp:
```diff
@@ -0,0 +1,61 @@
+#include "stdafx.h"
+#include "WeaponOffsetConfig.h"
+
+char* path = strcat
```
this should preferably be executed in the scope of the function requiring the file path, not globally

nukeulater replied:
also could probably use `std::filesystem` library here

nukeulater commented on xlive/H2MOD/Modules/WeaponOffsets/WeaponOffsetConfig/WeaponOffsetConfig.cpp:
```diff
@@ -0,0 +1,61 @@
+    {
+        if (defaultOffsets == false)
+        {
+            for (byte i = BattleRifle; i != Sniper + 1; i++)
```
weird and unusual way of looping through some array, please fix

nukeulater commented on xlive/H2MOD/Modules/WeaponOffsets/WeaponOffsetConfig/WeaponOffsetConfig.h:
```diff
@@ -0,0 +1,34 @@
+#include "Blam\Cache\TagGroups\weapon_definition.hpp"
+
+struct s_weapon_custom_offset {
+	const real_vector3d DefaultOffset;
```
variable names should start with lower case characters, upper reserved for class/struct member functions

nukeulater commented on xlive/H2MOD/Modules/WeaponOffsets/WeaponOffsetConfig/WeaponOffsetConfig.h:
```diff
@@ -0,0 +1,34 @@
+
+enum Weapons
+{
+	BattleRifle,
```
quite ambiguous enum elements here (e.g BattleRifle could be used in other parts of the code, not just within weapon offset modification context), and enum name should use `e_` prefix with lowercase name after

nukeulater commented on xlive/H2MOD/Modules/WeaponOffsets/WeaponOffsetConfig/WeaponOffsetConfig.h:
```diff
@@ -0,0 +1,34 @@
+};
+
+void ReadWeaponOffsetConfig(s_weapon_custom_offset* WeaponOffsets);
+void SaveWeaponOffsetConfig(const s_weapon_custom_offset customOffsets[], bool defaultOffsets);
```
prefer using pointers over arrays for function arguments, array in this case does decay to pointers but we should preferably be consistent across the codebase

nukeulater commented on xlive/H2MOD/Modules/WeaponOffsets/WeaponOffsets.cpp:
```diff
@@ -0,0 +1,218 @@
+		}
+		void MapLoad()
+		{
+			for (byte i = BattleRifle; i != Sniper + 1; i++)
```
same thing here, weird and unusual way of looping

nukeulater commented on xlive/H2MOD/Modules/WeaponOffsets/WeaponOffsetConfig/WeaponOffsetConfig.cpp:
```diff
@@ -0,0 +1,61 @@
+    file = fopen(path, "r");
+    if (file != NULL)
+    {
+        for (byte i = BattleRifle; i != Sniper + 1; i++)
```
weird and unusual way of looping through some array, please fix

nukeulater commented on xlive/H2MOD/Modules/WeaponOffsets/WeaponOffsets.cpp:
```diff
@@ -0,0 +1,218 @@
+			}
+		}
+
+		char* GetString(e_weapon_offsets_string string, const std::string& id)
```
if this function returns a string buffer that shouldn't be modified, it should return `const char*`. I noticed you just copied the code from another place expecting it to be fine. you might want to fix this there as well and remove the casting that removes the constness of the string buffers.

Berthalamew committed:
More Cleanup

- Add requested changes for Weapon Offset Code
- Cleanup tag fixes code
- Cleanup crosshair size code to use tagdefs

Berthalamew committed:
add requested changes

Berthalamew committed:
Update Project_Cartographer.vcxproj

Berthalamew committed:
Update WeaponOffsetConfig.cpp
